Strip out advice lines in docstrings

hlissner opened the pull request:
When a command is advised, these lines are prepended to its docstring:

![image](https://user-images.githubusercontent.com/510883/71452568-29307880-2754-11ea-8cff-805fb865c9bf.png)

When `which-key-show-docstrings` is non-nil, and which-key lists an advised command, we see those unhelpful lines rather than their documentation:

![image](https://user-images.githubusercontent.com/510883/71452543-000fe800-2754-11ea-88e8-dc67ff70c0b5.png)

This PR strips out these lines.

![image](https://user-images.githubusercontent.com/510883/71452690-136f8300-2755-11ea-985c-273a70fc76eb.png)

justbur commented:
Thanks. That regexp looks automatically generated. Can it be made easier to read using `regexp-opt` or something similar. Note that I am already using `regexp-opt` elsewhere?

hlissner commented:
Done.

- Switched out regexp literal for `regex-opt`
- Switched out `string-trim-left` (whose regexp argument wasn't introduced until 26.1) for `replace-regexp-in-string`
- Fixed type errors when commands have no documentation.

justbur commented:
Thanks, sorry for the delay in merging it. It slipped my mind.
